btas/generic/cp_als.h

```diff
@@ -0,0 +1,524 @@
+#ifndef BTAS_HAS_CP
+#define BTAS_HAS_CP
+
+/*!Canonical Product*/
+
+#include <algorithm>
+#include <iostream>
+#include <stdlib.h>
```

Comment:
should not use C header names when there are C++ equivalents: `stdlib.h` -> `cstdlib`